Tiled gallery: Improve image placeholder animation performance

sirreal opened the pull request:
## Proposed changes:

Change a loading animation to be applied conditionally to images as they load in the editor.

Before this change, the background-color animations applied to images always ran, indefinitely, in the editor.

This is an important performance improvement because the animations continued to run and cause significant browser rendering work, even though the background and animations were not visible. It was particularly noticeable on some recent Safari versions.

That's _especially_ bad on sites where the editor assets are loaded on the frontend. That is not typically the case, but there are some sites that do that.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Performance discussion: p1759329948724359-slack-C4GAQ900P

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

- Try adding images to an image block in the editor.
- As the image uploads, the previous behavior should be used (for blob URLs).
- When the images switch to their regular URL (after they've been uploaded), as the images load they should use the loading animation. The image containers should have an `is-loading` class.
- After the images are uploaded or if they fail (try setting offline in browser network tools) the images should not have a loading animation.

https://github.com/user-attachments/assets/5c66c5b8-9c2c-44ad-9d9b-4dce46ffbb9b

sirreal committed:
Simplify

sirreal committed:
changelog

sirreal committed:
check image.complete for initial state

sirreal committed:
check for complete in mount

sirreal committed:
Add once, passive to event handlers

sirreal committed:
Simplify and fix things

sirreal committed:
Simplify the loading animation

sgomes commented on projects/plugins/jetpack/extensions/blocks/tiled-gallery/editor.scss:
```diff
@@ -60,9 +56,9 @@
+		&.is-loading > img {
 			// Inspired by Calypso's placeholder mixin
-			animation: tiled-gallery-img-placeholder 1.6s ease-in-out infinite;
+			animation: tiled-gallery-img-placeholder 1.6s ease-in-out infinite alternate;
```
Nitpick: I think that if we want to keep the exact same effect, we need to adjust the speed, given that we effectively changed the frequency. Not sure it matters, though.

sirreal replied:
👍 Good spot. It needs to be halved.

[Different.](https://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0A%40keyframes%20anim%20%7B%0A0%25%20%7B%20background-color%3A%20red%3B%20%7D%0A50%25%20%7B%20background-color%3A%20blue%3B%20%7D%0A100%25%20%7B%20background-color%3A%20red%3B%20%7D%0A%7D%0A%40keyframes%20anim-alt%20%7B%0A0%25%20%7B%20background-color%3A%20red%3B%20%7D%0A100%25%20%7B%20background-color%3A%20blue%3B%20%7D%0A%7D%0Ali%20%7B%0A%20animation%3A%20anim%201.6s%20ease-in-out%20infinite%3B%0A%7D%0Ali%3Anth-of-type(2n)%20%7B%0A%20animation-name%3A%20anim-alt%3B%0A%20animation-direction%3A%20alternate%3B%0A%20%2F*%20animation-duration%3A%20.8s%3B%20*%2F%0A%7D%0A%3C%2Fstyle%3E%0A%3Col%3E%3Cli%3E%3Cli%3E)
[Same.](https://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Cstyle%3E%0A%40keyframes%20anim%20%7B%0A0%25%20%7B%20background-color%3A%20red%3B%20%7D%0A50%25%20%7B%20background-color%3A%20blue%3B%20%7D%0A100%25%20%7B%20background-color%3A%20red%3B%20%7D%0A%7D%0A%40keyframes%20anim-alt%20%7B%0A0%25%20%7B%20background-color%3A%20red%3B%20%7D%0A100%25%20%7B%20background-color%3A%20blue%3B%20%7D%0A%7D%0Ali%20%7B%0A%20animation%3A%20anim%201.6s%20ease-in-out%20infinite%3B%0A%7D%0Ali%3Anth-of-type(2n)%20%7B%0A%20animation-name%3A%20anim-alt%3B%0A%20animation-direction%3A%20alternate%3B%0A%20animation-duration%3A%20.8s%3B%0A%7D%0A%3C%2Fstyle%3E%0A%3Col%3E%3Cli%3E%3Cli%3E)

```suggestion
			animation: tiled-gallery-img-placeholder .8s ease-in-out infinite alternate;
```

sgomes commented on projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/edit.js:
```diff
@@ -26,8 +33,66 @@ class GalleryImageEdit extends Component {
+			if ( ! isComplete ) {
+				this.img.current.addEventListener( 'load', this.onImageLoadComplete, {
+					once: true,
+					passive: true,
```
Nitpick: I don't think `passive: true` will do anything here, since there's no default action, and therefore nothing that you could be opting out of interrupting. Still, it should be safe to keep.

sirreal replied:
👍 Seems like you're correct.
```suggestion
```

sgomes commented on projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/edit.js:
```diff
@@ -26,8 +33,66 @@ class GalleryImageEdit extends Component {
+				} );
+				this.img.current.addEventListener( 'error', this.onImageLoadComplete, {
+					once: true,
+					passive: true,
```
Same nitpick as above.

sirreal replied:
```suggestion
```

sgomes reviewed (dismissed):
LGTM, thank you @sirreal!

This looks good and tests well for me 👍

sirreal committed:
Fixes from code review

- The updated timing must be halved to match previous behavior.
- Passive does nothing in the case of these event handlers with no default action.

sirreal committed:
stylelint
